openedx/features/course_experience/views/course_home_messages.py

```diff
@@ -121,3 +135,66 @@ def register_course_home_messages(request, course, user_access, course_start_dat
                     course_start_date=course_start_data['course_start_date']
                 )
             )
+
+        is_already_verified = CourseEnrollment.is_enrolled_as_verified(request.user, course_key)
```

Comment:
I still need to review the details more closely, but it would help to break out functions for:
1. show_initial_course_goal_message()
2. get_initial_course_goal_message()

Also:
- Maybe this code should live in course_goals.py in this same directory.
- Maybe the new django app will be coursegoal_api and this will contain the UI?  I wouldn't doing anything here, but we should discuss. This is me thinking out loud and not convinced I know yet.
- The function `get_goal_text` below should move with this code.